File: qc-core/src/main/java/com/eager/questioncloud/core/domain/cart/implement/CartItemRemover.java
package com.eager.questioncloud.core.domain.cart.implement;

import com.eager.questioncloud.core.domain.cart.repository.CartItemRepository;
import com.eager.questioncloud.core.domain.payment.event.CompletedQuestionPaymentEvent;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@Component
@RequiredArgsConstructor
public class CartItemRemover {
    private final CartItemRepository cartItemRepository;

    @Transactional
    public void remove(List<Long> ids, Long userId) {
        cartItemRepository.deleteByIdInAndUserId(ids, userId);
    }

    @EventListener
    @Transactional
    public void clearCartItem(CompletedQuestionPaymentEvent event) {
        cartItemRepository.deleteByQuestionIdInAndUserId(event.getQuestionIds(), event.getQuestionPayment().getUserId());
    }
}

File: qc-core/qc-cart/qc-cart-api/src/main/kotlin/com/eager/questioncloud/cart/implement/CartItemRemover.kt
package com.eager.questioncloud.cart.implement

import com.eager.questioncloud.cart.repository.CartItemRepository
import org.springframework.stereotype.Component

@Component
class CartItemRemover(
    private val cartItemRepository: CartItemRepository,
) {
    fun removeCartItem(ids: List<Long>, userId: Long) {
        cartItemRepository.deleteByIdInAndUserId(ids, userId)
    }
}